internal/typename/gotypes/typesname.go

```diff
@@ -0,0 +1,137 @@
+package gotypes
+
+import (
+	"go/types"
+)
+
+// Of returns the fully-qualified name of the given type.
+func Of(t types.Type) string {
+	switch t := t.(type) {
+	case *types.Basic:
+		return t.String()
+	case *types.Pointer:
+		return t.String()
+	case *types.Slice:
+		return t.String()
+	case *types.Array:
+		return t.String()
+	case *types.Map:
+		return t.String()
+	case *types.Chan:
+		return buildChanName(t)
+	case *types.Interface:
+		return buildInterfaceName(t)
+	case *types.Named:
+		return t.String()
+	case *types.Struct:
+		return buildStructName(t)
+	case *types.Signature:
+		return "func" + buildFuncSignature(t)
+	default:
+		return t.String()
```

Comment:
Is it even possible to hit this?

Reply:
Don't know, TBH, but I suspect no as all the types are pretty much covered in the above cases. Should I just panic in `default` case?

Reply:
Yeah I think that's probably better.

Reply:
Should be fixed in https://github.com/dogmatiq/configkit/pull/115/commits/90f5f33d0321ceb32b1d58ce433d054227facfb5